spec/index.bs

```diff
@@ -875,26 +876,32 @@ A [=device=] has the following internal slots:
     ::
         The [=adapter=] from which this device was created.
 
-    : <dfn>\[[features]]</dfn>, of type [=list=]&lt;{{GPUFeatureName}}&gt;, readonly
+    : <dfn>\[[features]]</dfn>, of type [=ordered set=]&lt;{{GPUFeatureName}}&gt;, readonly
     ::
         The [=features=] which can be used on this device.
         No additional features can be used, even if the underlying [=adapter=] can support them.
 
-    : <dfn>\[[limits]]</dfn>, of type {{GPULimits}}, readonly
+    : <dfn>\[[limits]]</dfn>, of type [=supported limits=], readonly
     ::
         The limits which can be used on this device.
         No [=limit/better=] limits can be used, even if the underlying [=adapter=] can support them.
 </dl>
 
+Issue: [=device=] is already an internal concept. Having slots be marked as internal is redundant
+and adds extra `[[`/`]]` in spec text. Consider removing the brackets.
+
 <div algorithm>
     When <dfn dfn>a new device</dfn> |device| is created from [=adapter=] |adapter|
     with {{GPUDeviceDescriptor}} |descriptor|:
 
-      - Set |device|.{{device/[[adapter]]}} to |adapter|.
+    - Set |device|.{{device/[[adapter]]}} to |adapter|.
 
-      - Set |device|.{{device/[[features]]}} to |descriptor|.{{GPUDeviceDescriptor/features}}.
+    - Set |device|.{{device/[[features]]}} to the [=ordered set|set=] of values in
+        |descriptor|.{{GPUDeviceDescriptor/features}}.
 
-      - Set |device|.{{device/[[limits]]}} to |descriptor|.{{GPUDeviceDescriptor/limits}}.
+    - Let |device|.{{device/[[limits]]}} be a [=supported limits=] object with the default values.
+        For each (|key|, |value|) pair in |descriptor|.{{GPUDeviceDescriptor/limits}}, set the
+        member corresponding to |key| in |device|.{{device/[[limits]]}} to the value |value|.
```

Comment:
So, replacing GPULimits with `record<DOMString, GPUSize32>` does fix the problem of type conversions, which is handy, but it doesn't fix the problem that we no longer have the defaulting defined by IDL as we did before. But handling defaulting is not hard; done here.